tests/integration_rpc_server.rs

```diff
@@ -0,0 +1,55 @@
+#[cfg(test)]
+use assert_cmd::Command;
+
+#[test]
+fn deploy_and_invoke_contract_against_rpc_server() {
+    // This test assumes a fresh standalone network rpc server on port 8000
+
+    let mut cmd = Command::cargo_bin("soroban").unwrap();
+    // TODO: How to reuse the env variables between deploy and invoke
+    let deploy = cmd
+        .env("SOROBAN_RPC_URL", "http://localhost:8000/soroban/rpc")
+        .env(
+            "SOROBAN_SECRET_KEY",
+            "SC5O7VZUXDJ6JBDSZ74DSERXL7W3Y5LTOAMRF7RQRL3TAGAPS7LUVG3L",
+        )
+        .env(
+            "SOROBAN_NETWORK_PASSPHRASE",
+            "Standalone Network ; February 2017",
+        )
+        .args(&[
+            "deploy",
+            "--wasm=tests/fixtures/soroban_hello_world_contract.wasm",
```

Comment:
It looks like this test is dependent on a `soroban-examples` .wasm file. A while ago we removed dependencies on .wasm files in other repos because we found too often they would change and break and the approach was too brittle. Instead we've moved to generating .wasm files in the repos that require them. Take a look at the rs-soroban-env repo which has an implementation of this: https://github.com/stellar/rs-soroban-env/tree/main/soroban-test-wasms. The idea is any test vectors the CLI needs should be generated by code in the same repo as the test.

Reply:
@2opremio I'm actually in process of adding a .wasm fixture (via a crate) for #219. Would you like me to take the lead on the foundation for that? or do you want to do it? If you want to do it, I'll hold off on #219 until you're done.

Reply:
I agree it's better to track the source code than the wasm. But ... can't the contract source code also get outdated? Plus ... doesn't building the contract complicate the testing process too?

I can wait for #219 and see how it's done

Reply:
@2opremio I went ahead and used this pattern in the other PR. I'm still tidying it up, but that's the idea, take a look, wdyt?: https://github.com/stellar/soroban-cli/pull/219.

Reply:
I took a look, it seems simple enough. I am happy to migrate the test to that later on. Let's get this merged first if you don't mind.

Reply:
@sreuland probably, but given the largest risk with a change to the CLI is not changing the internal behavior, but the external visible behavior, I think the leaning into tests that exercise the CLI from the outside-in will have a larger impact on us providing a stable UX.

Reply:
agreed, the coverage is needed and executed as often as called for, but does that mean the external coverage and it's complexity has to be implemented in this repo? I [commented about de-coupling integration test and unit test concerns from repos a bit more](https://github.com/stellar/soroban-cli/pull/220#discussion_r998729023), if that aligns, maybe external testing can be implemented as e2e cases once in a dedicated 'qa' repo instead of variations of its complexity inside each repo. Still retain the ability to execute the tests as often as needed from 'qa' repo per different environments like GHA workflow(CI) or local per configs.

Reply:
The follow up to this is in https://github.com/stellar/soroban-cli/pull/222.